HPCC-21856 Filter popup displayed when not visible

miguelvazq opened the pull request:
Whenever a user clicks on DFUQueryWidget and they exit that widget prematurely the filter popup will still popup. This was also happening when searching through global search

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

miguelvazq commented:
@GordonSmith Please review.

HPCCSmoketest commented:
Process of PR-12441, label: HPCC-21856 starts now.
The reason of this test is: Did not build yet.
Commit ID: 1d1e87381661a34e55911ef345768da08dcfa886
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 1d1e87381661a34e55911ef345768da08dcfa886
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   36 sec (00:00:36) |  208 sec (00:03:28) |    0 sec (00:00:00) |   68 sec (00:01:08) |   17 sec (00:00:17) |    0 sec (00:00:00) |   16 sec (00:00:16) |  345 sec (00:05:45) |

GordonSmith commented on esp/src/eclwatch/DFUQueryWidget.js:
```diff
@@ -441,7 +441,14 @@ define([
                 this.initWorkunitsGrid();
-                this.checkIfWarning();
+               
+                !params.searchResults ? this.checkIfWarning() : null;
```
Should be a simple `if`.

richardkchapman commented:
@miguelvazq This should probably target 7.0.x

GordonSmith reviewed (changes requested):
I think the warning should only show once - but this looks like it will show every time user tabs to the page?

miguelvazq commented:
The warning will show if they come back to the tab and the filter is not set.

miguelvazq commented:
> @miguelvazq This should probably target 7.0.x



> @miguelvazq This should probably target 7.0.x

@richardkchapman there is a fix that @GordonSmith put in that looks like is in 7.2 which is why I added here.

miguelvazq commented:
@GordonSmith updated and comment added.

GordonSmith commented:
That fix originally went into: community_7.0.26-1:
![image](https://user-images.githubusercontent.com/966863/55965423-7cb67100-5c6e-11e9-95a9-946db54f2da0.png)

HPCCSmoketest commented:
Process of PR-12441, label: HPCC-21856 starts now.
The reason of this test is: New commit.
Commit ID: c84cf6939ec8389e20a07358d19ffd7e94d209cb
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: c84cf6939ec8389e20a07358d19ffd7e94d209cb
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   41 sec (00:00:41) |  211 sec (00:03:31) |    0 sec (00:00:00) |   66 sec (00:01:06) |   20 sec (00:00:20) |    0 sec (00:00:00) |   15 sec (00:00:15) |  353 sec (00:05:53) |

Commit:
HPCC-21856 Filter popup displayed when not visible

Whenever a user clicks on DFUQueryWidget and they exit that widget
prematurely the filter popup will still popup. This was also happening
when searching through global search

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexisrisk.com>

miguelvazq commented:
Changed base to 7.0.x

GordonSmith reviewed (approved):
Looks fine.

HPCCSmoketest commented:
Process of PR-12441, label: HPCC-21856 starts now.
The reason of this test is: New commit.
Commit ID: 8867b0e2d7b1afd28c0d64a299910eed37105a34
Estimated completion time is ~0.1 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 8867b0e2d7b1afd28c0d64a299910eed37105a34
Build: success
Build: success
Install HPCC Platform
HPCC Start: OK
HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   35 sec (00:00:35) |  176 sec (00:02:56) |    0 sec (00:00:00) |   62 sec (00:01:02) |   18 sec (00:00:18) |    0 sec (00:00:00) |   14 sec (00:00:14) |  305 sec (00:05:05) |
